### ISSUE TITLE
NVMeFix version 1.0.8 and 1.0.9 draining twice as much battery than version 1.0.7

### ISSUE THREAD

**Issue by siddiqq** [acidanthera/bugtracker#1773]
@vit9696 Hi, I'm sorry for posting again, but I've discovered something that I think will be important and helpful for you and other users. I hope you don't dismiss it as I've spent quite a long time troubleshooting this. I don't know if you remember but I posted about 3-4 months ago and had trouble with getting the debug log even with your guidance,  (https://github.com/acidanthera/bugtracker/issues/1620) so apologies for not posting one here.

So since my last post where I said the latest version of NVMeFix wasn't working properly on the latest Mac OS, I did some trial and error with a couple other versions of the kext and it seems 1.0.7 restores the battery life for me on my MBP Big Sur 11.5.2 which is great news, but some other people may not be aware of this and might just be installing the latest version of the kext without realizing the older version might be better.

Then I tried using 1.0.8 and 1.0.9 and confirmed they both drained the battery again. I took screen shots of the Energy Impact graph with all 3 versions installed and you can see that 1.0.7 has the graph perform normally with the line going to the bottom at idle, and giving me 9-10 hours, while the other 2 versions cause the idle line to remain higher and only give 4-5 hours.

1.0.7
<img width="209" alt="v107" src="https://user-images.githubusercontent.com/30996644/132089864-5755dbde-225b-4877-9be0-79482acd02f2.png">

1.0.8 
<img width="197" alt="v108" src="https://user-images.githubusercontent.com/30996644/132089872-d85c87cb-d2a7-4a26-b3dd-11059775394d.png">

1.0.9
<img width="197" alt="v109" src="https://user-images.githubusercontent.com/30996644/132089879-7a1cbb96-bed5-4b27-9e1f-b7c46fb750a6.png">

So it seems something broke after 1.0.7 with NVMeFix itself. Again, apologies for not exactly following the rules, its just an oddly specific question and you're the only person who knows how the kext works and how to fix it.

Cheers

**Comment by vit9696** [comment]
Hi, can you find the exact commit that breaks things? Use these builds, and tell me the last commit that drains little battery and the first one that drains a lot.
https://dortania.github.io/builds/?product=NVMeFix&viewall=true

Please make sure NVMeFix is loaded by calling kextstat.

**Comment by siddiqq** [comment]
> Hi, can you find the exact commit that breaks things? Use these builds, and tell me the last commit that drains little battery and the first one that drains a lot.
> https://dortania.github.io/builds/?product=NVMeFix&viewall=true
> 
> Please make sure NVMeFix is loaded by calling kextstat.

Hello,

So now I've got an even weirder thing going on.. I installed the first commit of 1.0.8, and it was draining the battery. Then I installed the last commit of 1.0.7 again to double check and it was also still draining, even though it wasn't before. Then I uninstalled it and reinstalled 1.0.7 again and the draining stopped.

I repeated this again, installing 1.0.8, then 1.0.7, and again it was still draining. Then I uninstalled and reinstalled again and the draining stopped. I checked the kextstat every time and it showed the kexts were loaded.

Any ideas why this is happening? Seems sometimes it installs correctly, and others it doesn't, even though its the exact same kext and the same steps I follow. I'm glad I at least got it to work again though.

Thanks

**Comment by vit9696** [comment]
macOS may do more or less I/O from boot to boot. Like indexing or backup. When there is I/O performed the consumed power may strongly increase. You can always check that you installed NVMeFix correctly by `kextstat` and inspecting `IORegistry` values. For now I will close this as there is no reliably reproducible issue. Please continue providing us with the details (posting here).

**Comment by siddiqq** [comment]
> macOS may do more or less I/O from boot to boot. Like indexing or backup. When there is I/O performed the consumed power may strongly increase. You can always check that you installed NVMeFix correctly by `kextstat` and inspecting `IORegistry` values. For now I will close this as there is no reliably reproducible issue. Please continue providing us with the details (posting here).

Hi,

So I've done some more troubleshooting and it seems to be that when I do the initial install of 1.0.7, as in I use the command  "sudo cp -R ~/Downloads/NVMeFix.kext /Library/Extensions", I get a prompt to restart my Mac in order to install the kext. Then after the reboot there is no battery drain and I get good 9-10 hours. 

However if I restart again myself, the battery drain comes back and I get 4-5 hours, even though the exact same kexts are loaded and untouched. I repeated this a few more times by uninstalling, restarting, the installing again and get the same results. 

So I think something is loading into the initial boot up right after the installation, and not loading when the Mac is restarted manually. Also, you are correct in that the Mac does some indexing for about 1-2 minutes after a reboot, but even when its complete, the battery drain comes remains.
---------------
Okay so I did one more thing after writing that up, I installed Lilu 1.5.3 as I thought I'd try a version that released around the same time as NVMeFix 1.0.7, and after rebooting, there was no battery drain, same as the previous tests. BUT this time when I restarted manually, the battery drain was still gone!

So at this point I think its either : My setup is causing something random during the boot up of the kexts and its too difficult to pinpoint what the issue is since there's too many variables (my SSD, MacBook model, software, etc), OR its the kexts themselves that prefer to work on certain OS versions but not others.

Either way I'm glad I at least found a good combination of fixes. My only worry is if this will still work on future versions of MacOS Monterey, etc. We'll see how it goes :)
